docs/messaging.rst

```diff
@@ -1052,9 +1064,26 @@ Message type: ``status``::
         execution_state : ('busy', 'idle', 'starting')
     }
 
+When a kernel receives a request and begins processing it,
+it shall immediately publish a status message with ``execution_state: 'busy'``.
+When that kernel has completed processing the request
+and has finished publishing associated IOPub messages, if any,
+it shall publish a status message with ``execution_state: 'idle'``.
+Thus, the outputs associated with a given execution shall generally arive
+between the idle and busy status messages associated with a given request.
+
+.. note::
+
+    There is a caveat for asynchronous output,
```

Comment:
**A caveat for asynchronous output**

Asynchronous output may be produced after the kernel has sent the idle status message. The kernel sends an idle message when the kernel has completed processing the request. The handling of asynchronous output messages is currently undefined in the specification (reference link to spec??). The currently implemented behavior is the Jupyter Notebook ...